sdk/cosmos/azure-cosmos/azure/cosmos/container.py

```diff
@@ -119,7 +121,6 @@ def read(
         # type: (...) -> Dict[str, Any]
         """Read the container properties.
 
-        :param populate_query_metrics: Enable returning query metrics in response headers.
```

Comment:
If you're keen.... we could go further than this and unbreakingly remove it from the method signature.....
Something like:
```python
    @overload
    def read(
        self,
        *,
        populate_partition_key_range_statistics: Optional[bool] = None,
        populate_quota_info: Optional[bool] = None,
        **kwargs
    )

    def read(
        self,
        *args,
        **kwargs
    ):
        populate_query_metrics = args[0] if args else None or kwargs.pop('populate_query_metrics', None)
        if populate_query_metrics:
            ...  # warning
```

Reply:
I don't think we need to go that far since regardless the option can still be passed in and never did anything to begin with. This is a good point of reference for future refactoring though, thank you! :)

Reply:
That's true - the reason I raised it is that currently the API reference documentation will display a positional keyword parameter that has no accompanying documentation. Which may confuse.
What we _could_ do - is leave the param documentation in, but just describe it as a deprecated no-op.

But yes - adding the fancier overloads are a non-breaking refactor we could do later before the GA of the async APIs to clean up the old sync APIs to match any 'improvements' we want to add in the async version.

Reply:
I see what you mean... how would we accomplish the work for this item then though? From our initial conversation, we had concluded on deleting the documentation for this option and marking it as deprecated if users passed it in. Does this mean we now want to still provide this documentation (even though the option is useless and marked to be removed)?